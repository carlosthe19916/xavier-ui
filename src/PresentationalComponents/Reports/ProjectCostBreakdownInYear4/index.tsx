import ProjectCostBreakdownInYear4 from './ProjectCostBreakdownInYear4';

export default ProjectCostBreakdownInYear4;
