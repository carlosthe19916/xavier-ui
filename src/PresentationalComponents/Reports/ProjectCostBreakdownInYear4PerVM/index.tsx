import ProjectCostBreakdownInYear4PerVM from './ProjectCostBreakdownInYear4PerVM';

export default ProjectCostBreakdownInYear4PerVM;
