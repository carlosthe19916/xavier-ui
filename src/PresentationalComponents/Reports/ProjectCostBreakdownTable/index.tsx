import ProjectCostBreakdownTable from './ProjectCostBreakdownTable';

export default ProjectCostBreakdownTable;
