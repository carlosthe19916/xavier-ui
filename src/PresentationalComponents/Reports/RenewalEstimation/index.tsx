import RenewalEstimation from './RenewalEstimation';

export default RenewalEstimation;
