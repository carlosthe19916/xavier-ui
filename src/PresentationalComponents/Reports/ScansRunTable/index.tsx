import ScansRunTable from './ScansRunTable';

export default ScansRunTable;
