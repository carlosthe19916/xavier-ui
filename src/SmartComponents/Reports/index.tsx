import React, { createRef } from 'react';
import { bindActionCreators } from 'redux';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import {
    TableToolbar
} from '@redhat-cloud-services/frontend-components';
import {
    Button,
    ToolbarGroup,
    ToolbarItem,
    Bullseye,
    EmptyStateIcon,
    EmptyStateBody,
    EmptyStateVariant,
    EmptyState,
    Title
} from '@patternfly/react-core';
import {
    Table,
    TableHeader,
    TableBody,
    TableGridBreakpoint,
    IRow,
    ICell
} from '@patternfly/react-table';
import ReportListPage from '../../PresentationalComponents/ReportListPage/ReportListPage';
import LoadingState from '../../PresentationalComponents/LoadingState/LoadingState';
import  * as reportActions from '../../actions/ReportActions';
import * as uploadActions from '../../actions/UploadActions';
import { Report } from '../../models';
import { GlobalState } from '../../models/state';
import { RouterGlobalProps } from '../../models/router';
import './ReportList.scss';
import Dropzone from 'react-dropzone';
import { CubesIcon } from '@patternfly/react-icons';

interface StateToProps {
    total: number;
    error: string | null;
    loading: boolean;
    reports: Report[];

    file: File | null
}

interface DispatchToProps {
    fetchReports: () => any;
    selectUploadFile: (file: File) => any;
}

interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

interface State {
    columns: Array<ICell | String>;
    rows: Array<IRow | Array<String>>
};

export class Reports extends React.Component<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {
            columns: [
                { title: 'Report Id', props: {}},
                { title: 'Customer Id', props: {}},
                {
                    title: 'File name',
                    props: {
                        className: 'pf-u-text-align-center'
                    }
                },
                ''
            ],
            rows: []
        };
        this.onFileSelected = this.onFileSelected.bind(this);
    }

    componentDidMount(): void {
        this.refreshData();
    }

    refreshData(): void {
        this.props.fetchReports().then(() =>
            this.filtersInRowsAndCells()
        );
    }

    filtersInRowsAndCells(): void {
        const reports: Report[] = Object.values(this.props.reports);

        let rows: any[][] = [];
        if (reports.length > 0) {
            rows = reports.map(({ id, customerId, fileName }) => (
                [
                    { title: id },
                    { title: customerId },
                    { title: fileName },
                    {
                        title: <Button variant='primary' component={ Link } to={ `/reports/${id}` }>View</Button>
                    }
                ]
            ));
        }

        this.setState({ rows });
    }

    onFileSelected(files: File[]): void {
        this.props.selectUploadFile(files[0]);
        this.props.history.push('/reports/upload');
    };

    noResults() {
        const dropzoneRef: any = createRef();
        const openFileDialog = () => {
            if (dropzoneRef.current) {
                dropzoneRef.current.open();
            }
        };

        return (
            <Bullseye>
                <EmptyState variant={ EmptyStateVariant.full }>
                    <p>
                        <EmptyStateIcon icon={ CubesIcon } />
                    </p>
                    <Title size="lg">Let Red Hat Migration Analytics suggest ways to optimize your environment</Title>
                    <EmptyStateBody>
                                See how you can optimize your virtual environment by
                                uploading a Red Hat CloudForms generated inventory file
                                and then letting Red Hat Migration Analytics create
                                reports showing how you can save money and optimize
                                workloads - whether by migrating virtual machines or
                                migrating application to RHEL.
                    </EmptyStateBody>
                    <Dropzone
                        onDrop={ this.onFileSelected }
                        ref={ dropzoneRef }
                        noClick noKeyboard
                        multiple={ false }
                        accept={ [ 'application/zip', 'application/json' ] }
                    >
                        { ({ getRootProps, getInputProps }) => {
                            return (
                                <div { ...getRootProps({ className: 'dropzone' }) }>
                                    <input { ...getInputProps() } />
                                    <Button
                                        type="button"
                                        variant="primary"
                                        onClick={ openFileDialog }>
                                        Upload
                                    </Button>
                                </div>
                            );
                        } }
                    </Dropzone>
                </EmptyState>
            </Bullseye>
        );
    }

    resultsTable() {
        const { rows, columns } = this.state;

        return (
            <React.Fragment>
                { <TableToolbar>
                    <ToolbarGroup>
                        <ToolbarItem>
                            <Button component={ Link } to={ '/upload' }>Upload</Button>
                        </ToolbarItem>
                    </ToolbarGroup>
                </TableToolbar> }
                <Table aria-label='Reports list'
                    rows={ rows }
                    cells={ columns }
                    gridBreakPoint={ TableGridBreakpoint.gridMd } >
                    <TableHeader />
                    <TableBody />
                </Table>
            </React.Fragment>
        );
    }

    render() {
        const { loading, total } = this.props;

        return (
            <ReportListPage
                title='Reports'
                showBreadcrumb={ false }>
                <LoadingState
                    loading={ loading }
                    placeholder={ '' } >
                    { total > 0 ? this.resultsTable() : this.noResults() }
                </LoadingState>
            </ReportListPage>
        );
    }
}

const mapStateToProps = (state: GlobalState)  => {
    let {
        reportState: {
            reports,
            loading,
            error,
            total
        },
        uploadState: {
            file
        }
    } = state;
    return {
        reports,
        loading,
        error,
        total,
        file
    };
};

const mapDispatchToProps = (dispatch: any) =>
    bindActionCreators({
        fetchReports: reportActions.fetchReports,
        selectUploadFile: uploadActions.selectUploadFile
    }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Reports));
