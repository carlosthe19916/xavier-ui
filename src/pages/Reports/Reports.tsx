import React from 'react';
import { Link } from 'react-router-dom';
import {
    TableToolbar,
    SkeletonTable
} from '@redhat-cloud-services/frontend-components';
import {
    Button,
    ToolbarGroup,
    ToolbarItem,
    ButtonVariant,
    Pagination,
    InputGroup,
    TextInput,
    Form,
    EmptyState,
    EmptyStateVariant,
    EmptyStateIcon,
    Title,
    EmptyStateBody,
    Card,
    CardBody
} from '@patternfly/react-core';
import {
    Table,
    TableHeader,
    TableBody,
    TableGridBreakpoint,
    IRow,
    ICell
} from '@patternfly/react-table';

import './Reports.scss';
import { Report } from '../../models';
import { RouterGlobalProps } from '../../models/router';
import ReportsPage from '../../PresentationalComponents/ReportsPage';
import * as deleteActions from '../../actions/DialogDeleteActions';
import {
    SearchIcon,
    OkIcon,
    ErrorCircleOIcon,
    InProgressIcon
} from '@patternfly/react-icons';
import debounce from 'lodash/debounce';
import { Formik } from 'formik';
import { ObjectFetchStatus } from '../../models/state';

interface StateToProps {
    reports: {
        total: number;
        items: Report[];
    };
    reportsFetchStatus: ObjectFetchStatus;
}

interface DispatchToProps {
    fetchReports: (page: number, perPage: number, filterText: string) => any;
    deleteReport: (id: number, name: string) => any;
    showDeleteDialog: typeof deleteActions.openModal;
    closeDeleteDialog: typeof deleteActions.closeModal;
}

export interface Props extends StateToProps, DispatchToProps, RouterGlobalProps {
};

export interface State {
    filterText: string;
    page: number;
    perPage: number;
    columns: Array<ICell | string>;
    rows: Array<IRow | string[]>;
    isFirstFetchReportsCall: boolean;
};

const PULL_INTERVAL_TIME = 5000;

class Reports extends React.Component<Props, State> {

    public pullTimer: any;

    public changePage = debounce(() => {
        this.refreshData();
    }, 800);

    constructor(props: Props) {
        super(props);
        this.state = {
            filterText: '',
            page: 1,
            perPage: 10,
            columns: [
                'Name',
                'Status',
                {
                    title: '',
                    props: {
                        className: 'pf-u-text-align-center'
                    }
                }
            ],
            rows: [],
            isFirstFetchReportsCall: true
        };
    }

    // React lyfe cycle methods

    public componentDidMount() {
        this.refreshData();
        this.startTimer(this.refreshData);
    }

    public componentDidUpdate() {
        // If it is the first time fetching reports and there are no reports
        // then redirect to /no-reports page.
        const { reportsFetchStatus, reports } = this.props;
        const { isFirstFetchReportsCall } = this.state;
        if (reports.total === 0 && isFirstFetchReportsCall && reportsFetchStatus.status === 'complete') {
            this.props.history.push('/no-reports');
        }
    }

    public componentWillUnmount() {
        this.stopTimer();
    }

    // Pull timer config

    public startTimer = (callback: (...args: any[]) => void) => {
        if (this.pullTimer) {
            clearInterval(this.pullTimer);
        }

        this.pullTimer = setInterval(callback, PULL_INTERVAL_TIME);
    };

    public stopTimer = () => {
        clearInterval(this.pullTimer);
    };

    //

    public handleDeleteReport = (report: Report) => {
        const { deleteReport, showDeleteDialog, closeDeleteDialog } = this.props;

        showDeleteDialog({
            name: report.fileName,
            type: 'report',
            onDelete: () => {
                deleteReport(report.id, report.fileName).then(() => {
                    closeDeleteDialog();
                    this.refreshData();
                });
            },
            onCancel: () => {
                closeDeleteDialog();
            }
        });
    };

    public renderReportStatus = (report: Report) => {
        switch (report.status) {
            case 'CREATED':
                return <p><OkIcon className="success" /> Report created - { new Date(report.creationDate).toUTCString() }</p>;
            case 'FAILED':
                return <p><ErrorCircleOIcon className="error" /> Report failed - { new Date(report.creationDate).toUTCString() }</p>;
            case 'IN_PROGRESS':
                return <p><InProgressIcon className="progress" /> Analyzing the upload file</p>;
            default:
                return null;
        }
    };

    public renderReportActions = (report: Report) => {
        const onDelete = (_event: any): void => {
            this.handleDeleteReport(report);
        };

        switch (report.status) {
            case 'CREATED':
                return <Button variant={ ButtonVariant.secondary } onClick={ onDelete }>Delete</Button>;
            case 'FAILED':
                return <Button variant={ ButtonVariant.secondary } onClick={ onDelete }>Delete</Button>;;
            case 'IN_PROGRESS':
                return '';
            default:
                return null;
        }
    };

    public filtersInRowsAndCells(): void {
        const reports: Report[] = this.props.reports.items ? Object.values(this.props.reports.items) : [];

        let rows: any[][] = [];
        if (reports.length > 0) {
            rows = reports.map((report) => (
                [
                    {
                        title: <Link to={ `/reports/${report.id}` }>{ report.fileName }</Link>
                    },
                    {
                        title: this.renderReportStatus(report)
                    },
                    {
                        title: this.renderReportActions(report)
                    }
                ]
            ));
        }

        this.setState({ rows });
    }

    public refreshData = (page: number = this.state.page, perPage: number = this.state.perPage, filterText: string = this.state.filterText) => {
        this.props.fetchReports(page, perPage, filterText).then(() => {
            this.filtersInRowsAndCells();

            // Change to false to indicate that
            const { isFirstFetchReportsCall } = this.state;
            if (isFirstFetchReportsCall) {
                this.setState({ isFirstFetchReportsCall: false });
            }
        });
    }

    public onPageChange = (event: any, page: number, shouldDebounce: boolean) => {
        this.setState({ page });
        if (shouldDebounce) {
            this.changePage();
        } else {
            this.refreshData(page);
        }
    };

    public onSetPage = (event: any, page: number) => {
        return event.target.className === 'pf-c-form-control' || this.onPageChange(event, page, false);
    };

    public onPageInput = (event: any, page: number) => {
        return this.onPageChange(event, page, true);
    };

    public onPerPageSelect = (_event: any, perPage: number) => {
        let page = this.state.page;
        const total = this.props.reports.total;

        // If current page and perPage would request data beyond total, show last available page
        if (page * perPage > total) {
            page = Math.floor(total / perPage) + 1;
        }

        this.setState({ page, perPage });
        this.refreshData(page, perPage);
    };

    public renderNoResults() {
        return (
            <React.Fragment>
                <Card>
                    <CardBody>
                        <EmptyState variant={ EmptyStateVariant.full }>
                            <EmptyStateIcon icon={ SearchIcon } />
                            <Title headingLevel="h5" size="lg">No results found</Title>
                            <EmptyStateBody>
                                No results match the search criteria
                            </EmptyStateBody>
                        </EmptyState>
                    </CardBody>
                </Card>
            </React.Fragment>
        );
    }

    public renderResultsTable() {
        const { rows, columns } = this.state;

        return (
            <Table aria-label='Reports list'
                rows={ rows }
                cells={ columns }
                gridBreakPoint={ TableGridBreakpoint.gridMd } >
                <TableHeader />
                <TableBody />
            </Table>
        );
    }

    public handleSearchSubmit = (values: any) => {
        const filterText: string = values.filterText;
        this.setState({
            filterText: filterText.trim()
        });
        this.refreshData();
    };

    public renderSearchBox = () => {
        // Always return empty because every value is valid
        const searchBoxValidation = () => {
            return {};
        }

        return (
            <Formik
                initialValues={ { filterText: '' } }
                validate={ searchBoxValidation }
                onSubmit={ this.handleSearchSubmit }
            >
                {
                    ({
                        values,
                        handleChange,
                        handleBlur,
                        handleSubmit
                    }) =>
                        {
                            const customHandleChange = (_value: any, event: any) => {
                                handleChange(event);
                            };

                            return (
                                <Form onSubmit={ handleSubmit }>
                                    <InputGroup>
                                        <TextInput
                                            type="search"
                                            id="filterText"
                                            name="filterText"
                                            aria-label="search text input"
                                            onChange={ customHandleChange }
                                            onBlur={ handleBlur }
                                            value={ values.filterText }
                                            placeholder="Filter by name..."/>
                                        <Button type="submit" variant={ ButtonVariant.tertiary } aria-label="search button for search input">
                                            <SearchIcon />
                                        </Button>
                                    </InputGroup>
                                </Form>
                            );
                        }
                }
            </Formik>
        );
    };

    public render() {
        const { isFirstFetchReportsCall, page, perPage } = this.state;
        const { total } = this.props.reports;

        if (isFirstFetchReportsCall) {
            return (
                <ReportsPage>
                    <SkeletonTable colSize={ 3 } rowSize={ 10 }/>
                </ReportsPage>
            );
        }

        return (
            <ReportsPage>
                <TableToolbar className="pf-u-justify-content-space-between">
                    <ToolbarGroup>
                        <ToolbarItem className="pf-u-mr-xl">{ this.renderSearchBox() }</ToolbarItem>
                        <ToolbarItem className="pf-u-mr-md">
                            <Link to={ '/reports/upload' } className="pf-c-button pf-m-primary">Create</Link>
                        </ToolbarItem>
                    </ToolbarGroup>
                    <ToolbarGroup>
                        <ToolbarItem>
                            <Pagination
                                itemCount={ total }
                                perPage={ perPage }
                                page={ page }
                                onSetPage={ this.onSetPage }
                                onPageInput={ this.onPageInput }
                                onPerPageSelect={ this.onPerPageSelect }
                            />
                        </ToolbarItem>
                    </ToolbarGroup>
                </TableToolbar>
                { (total > 0 ? this.renderResultsTable() : this.renderNoResults()) }
            </ReportsPage>
        );
    }
}

export default Reports;
