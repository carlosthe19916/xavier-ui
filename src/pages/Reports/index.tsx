import { connect } from 'react-redux';
import Reports from './Reports';
import { withRouter } from 'react-router';
import { GlobalState } from '../../models/state';
import  * as reportActions from '../../actions/ReportActions';
import  * as dialogDeleteActions from '../../actions/DialogDeleteActions';

const mapStateToProps = (state: GlobalState)  => {
    const {
        reportState: {
            reports,
            loading,
            error,
            total
        }
    } = state;
    return {
        reports,
        loading,
        error,
        total
    };
};

const mapDispatchToProps = {
    fetchReports: reportActions.fetchReports,
    deleteReport: reportActions.deleteReport,
    showDeleteDialog: dialogDeleteActions.openModal,
    closeDeleteDialog: dialogDeleteActions.closeModal
};

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(Reports)
);
