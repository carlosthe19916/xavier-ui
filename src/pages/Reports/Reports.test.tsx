import React from "react";
import { shallow } from 'enzyme';
import Reports, { Props } from './Reports';

const props: Props = {
  reports: {
    total: 2,
    items: [
      {
        id: 36,
        customerId: "123456",
        fileName: "file1.json",
        numberOfHosts: 254,
        totalDiskSpace: 5871365,
        totalPrice: 1200,
        creationDate: 123654565464,
        status: 'IN_PROGRESS'
      },
      {
        id: 37,
        customerId: "654321",
        fileName: "file2.json",
        numberOfHosts: 574,
        totalDiskSpace: 5412584,
        totalPrice: 1800,
        creationDate: 123654565464,
        status: 'IN_PROGRESS'
      }
    ]
  },
  reportsFetchStatus: {
    error: null,
    status: 'complete'
  },

  fetchReports: () => ({
    then: (fn: Function) => fn()
  }),
  deleteReport: jest.fn(),
  showDeleteDialog: jest.fn(),
  closeDeleteDialog: jest.fn(),

  history: {
    push: jest.fn()
  },
  location: null,
  match: null
};

describe("Reports should render list depending of TOTAL value", () => {
  it("expect to render empty list even though is not fetching and items.length > 0. Using Total = 0", () => {
    const emptyProps: Props = {
      ...props,
      reports: {
        ...props.reports,
        total: 0
      },
      reportsFetchStatus: {
        ...props.reportsFetchStatus,
        status: 'inProgress'
      }
    };

    const wrapper = shallow(<Reports {...emptyProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expect to render list even though is fetching. Using Total > 0", () => {
    const listProps: Props = {
      ...props,
      reports: {
        ...props.reports,
        total: props.reports.items.length
      },
      reportsFetchStatus: {
        ...props.reportsFetchStatus,
        status: 'inProgress'
      }
    };
    const wrapper = shallow(<Reports {...listProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
