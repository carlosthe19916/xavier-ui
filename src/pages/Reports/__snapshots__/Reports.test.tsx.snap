// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reports should render list depending of TOTAL value expect to render empty list even though is not fetching and items.length > 0. Using Total = 0 1`] = `
<ReportsPage>
  <TableToolbar
    className="pf-u-justify-content-space-between"
    isFooter={false}
  >
    <ToolbarGroup>
      <ToolbarItem
        className="pf-u-mr-xl"
      >
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "filterText": "",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          validate={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
        >
          <Component />
        </Formik>
      </ToolbarItem>
      <ToolbarItem
        className="pf-u-mr-md"
      >
        <Link
          className="pf-c-button pf-m-primary"
          replace={false}
          to="/reports/upload"
        >
          Create
        </Link>
      </ToolbarItem>
    </ToolbarGroup>
    <ToolbarGroup>
      <ToolbarItem>
        <Pagination
          itemCount={0}
          onPageInput={[Function]}
          onPerPageSelect={[Function]}
          onSetPage={[Function]}
          page={1}
          perPage={10}
        />
      </ToolbarItem>
    </ToolbarGroup>
  </TableToolbar>
  <Card>
    <CardBody>
      <EmptyState
        variant="full"
      >
        <EmptyStateIcon
          icon={[Function]}
        />
        <Title
          headingLevel="h5"
          size="lg"
        >
          No results found
        </Title>
        <EmptyStateBody>
          No results match the search criteria
        </EmptyStateBody>
      </EmptyState>
    </CardBody>
  </Card>
</ReportsPage>
`;

exports[`Reports should render list depending of TOTAL value expect to render list even though is fetching. Using Total > 0 1`] = `
<ReportsPage>
  <TableToolbar
    className="pf-u-justify-content-space-between"
    isFooter={false}
  >
    <ToolbarGroup>
      <ToolbarItem
        className="pf-u-mr-xl"
      >
        <Formik
          enableReinitialize={false}
          initialValues={
            Object {
              "filterText": "",
            }
          }
          isInitialValid={false}
          onSubmit={[Function]}
          validate={[Function]}
          validateOnBlur={true}
          validateOnChange={true}
        >
          <Component />
        </Formik>
      </ToolbarItem>
      <ToolbarItem
        className="pf-u-mr-md"
      >
        <Link
          className="pf-c-button pf-m-primary"
          replace={false}
          to="/reports/upload"
        >
          Create
        </Link>
      </ToolbarItem>
    </ToolbarGroup>
    <ToolbarGroup>
      <ToolbarItem>
        <Pagination
          itemCount={2}
          onPageInput={[Function]}
          onPerPageSelect={[Function]}
          onSetPage={[Function]}
          page={1}
          perPage={10}
        />
      </ToolbarItem>
    </ToolbarGroup>
  </TableToolbar>
  <Table
    aria-label="Reports list"
    borders={true}
    cells={
      Array [
        "Name",
        "Status",
        Object {
          "props": Object {
            "className": "pf-u-text-align-center",
          },
          "title": "",
        },
      ]
    }
    className=""
    contentId="expanded-content"
    dropdownDirection="down"
    dropdownPosition="right"
    expandId="expandable-toggle"
    gridBreakPoint="grid-md"
    onCollapse={null}
    onExpand={null}
    role="grid"
    rowLabeledBy="simple-node"
    rows={
      Array [
        Array [
          Object {
            "title": <Link
              replace={false}
              to="/reports/36"
            >
              file1.json
            </Link>,
          },
          Object {
            "title": <p>
              <InProgressIcon
                className="progress"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
                title={null}
              />
               Analyzing the upload file
            </p>,
          },
          Object {
            "title": "",
          },
        ],
        Array [
          Object {
            "title": <Link
              replace={false}
              to="/reports/37"
            >
              file2.json
            </Link>,
          },
          Object {
            "title": <p>
              <InProgressIcon
                className="progress"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
                title={null}
              />
               Analyzing the upload file
            </p>,
          },
          Object {
            "title": "",
          },
        ],
      ]
    }
    variant={null}
  >
    <TableHeader
      className=""
    />
    <TableBody
      className=""
      onRowClick={[Function]}
      rowKey="id"
    />
  </Table>
</ReportsPage>
`;
